cron/centAcl.php

```diff
@@ -130,45 +131,52 @@ function programExit($msg)
     $resourceCache = array();
 
     /** **********************************************
-     * Sync ACL with ldap
+     * Sync ACL with each ldap depending on last sync hour and own synchronization interval
      */
-    $queryOptions = "SELECT `key`, `value` FROM `options` WHERE `key` IN ('ldap_auth_enable', 'ldap_last_acl_update')";
-    $res = $pearDB->query($queryOptions);
-    while ($row = $res->fetch()) {
-        switch ($row['key']) {
-            case 'ldap_auth_enable':
-                $ldap_enable = $row['value'];
-                break;
-            case 'ldap_last_acl_update':
-                $ldap_last_update = $row['value'];
-                break;
+    $ldapConf = $pearDB->query(
+        "SELECT auth.ar_id, auth.ar_sync_base_date, info.ari_value
+        FROM auth_ressource auth INNER JOIN auth_ressource_info info ON auth.ar_id = info.ar_id
+        WHERE auth.ar_enable = '1' AND info.ari_name = 'ldap_sync_interval'"
+    );
+    $reSync = false;
+    while ($ldapRow = $ldapConf->fetch()) {
+        $currentTime = time();
+        if ($ldapRow['ar_sync_base_date'] + 3600 * $ldapRow['ari_value'] <= $currentTime) {
+            $reSync = true;
+            $updateSyncTime = $pearDB->prepare(
+                "UPDATE auth_ressource SET ar_sync_base_date = :currentTime
+                WHERE ar_id = :arId"
+            );
+            $updateSyncTime->bindValue(':currentTime', $currentTime, \PDO::PARAM_INT);
+            $updateSyncTime->bindValue(':arId', (int)$ldapRow['ar_id'], \PDO::PARAM_INT);
+            $updateSyncTime->execute();
         }
-    }
 
-    /** ********************************************
-     * If the ldap is enable and the last check
-     * is greater than the update period
-     */
-    if ($ldap_enable == 1 && $ldap_last_update < (time() - LDAP_UPDATE_PERIOD)) {
+    }
+    if ($reSync === true) {
+        // @TODO : Synchronize LDAP with contacts data in background to avoid it at login
         $cgObj->syncWithLdap();
     }
 
+
     /** **********************************************
      * Remove data from old groups (deleted groups)
      */
-    $aclGroupToDelete = "SELECT DISTINCT acl_group_id FROM $centreonDbName.acl_groups WHERE acl_group_activate = '1'";
-    $aclGroupToDelete2 = "SELECT DISTINCT acl_group_id FROM $centreonDbName.acl_res_group_relations";
-    $pearDBO->query("DELETE FROM centreon_acl WHERE group_id NOT IN ($aclGroupToDelete)");
-    $pearDBO->query("DELETE FROM centreon_acl WHERE group_id NOT IN ($aclGroupToDelete2)");
+    $aclGroupToDelete = "SELECT DISTINCT acl_group_id " .
+        "FROM " . $centreonDbName . ".acl_groups WHERE acl_group_activate = '1'";
+    $aclGroupToDelete2 = "SELECT DISTINCT acl_group_id FROM " . $centreonDbName . ".acl_res_group_relations";
+    $pearDBO->query("DELETE FROM centreon_acl WHERE group_id NOT IN (" . $aclGroupToDelete . ")");
```

Comment:
Don't forget to create a db transaction when there is more than one update request.